src/bp-members/classes/class-bp-signup.php

```diff
@@ -1084,4 +980,161 @@ public static function delete( $signup_ids = array() ) {
+
+		$to_resend = self::get(
+			array(
+				'include' => wp_parse_id_list( $signup_ids ),
+			)
+		);
+
+		$signups = $to_resend['signups'];
+
+		if ( ! $signups ) {
+			return array();
+		}
+
+		$result = array();
+
+		/**
+		 * Fires before activation emails are resent.
+		 *
+		 * @since 2.0.0
+		 *
+		 * @param array $signup_ids Array of IDs to resend activation emails to.
+		 */
+		do_action( 'bp_core_signup_before_resend', $signup_ids );
+
+		foreach ( $signups as $signup ) {
+
+			$meta               = $signup->meta;
+			$meta['sent_date']  = current_time( 'mysql', true );
+			$meta['count_sent'] = $signup->count_sent + 1;
+
+			// Send activation email.
+			if ( is_multisite() ) {
+				// Should we send the user or blog activation email?
+				if ( ! empty( $signup->domain ) || ! empty( $signup->path ) ) {
+					wpmu_signup_blog_notification( $signup->domain, $signup->path, $signup->title, $signup->user_login, $signup->user_email, $signup->activation_key, $meta );
+				} else {
+					wpmu_signup_user_notification( $signup->user_login, $signup->user_email, $signup->activation_key, $meta );
+				}
+			} else {
+
+				// Check user status before sending email.
+				$user_id = email_exists( $signup->user_email );
+
+				if ( ! empty( $user_id ) && 2 !== self::check_user_status( $user_id ) ) {
+
+					// Status is not 2, so user's account has been activated.
+					$result['errors'][ $signup->signup_id ] = array( $signup->user_login, esc_html__( 'the sign-up has already been activated.', 'buddypress' ) );
+
+					// Repair signups table.
+					self::validate( $signup->activation_key );
+
+					continue;
+
+					// Send the validation email.
+				} else {
+					$salutation = $signup->user_login;
+					if ( bp_is_active( 'xprofile' ) && isset( $meta[ 'field_' . bp_xprofile_fullname_field_id() ] ) ) {
+						$salutation = $meta[ 'field_' . bp_xprofile_fullname_field_id() ];
+					}
+
+					bp_core_signup_send_validation_email( false, $signup->user_email, $signup->activation_key, $salutation );
+				}
+			}
+
+			// Update metas.
+			$result['resent'][] = self::update(
+				array(
+					'signup_id' => $signup->signup_id,
+					'meta'      => $meta,
+				)
+			);
+		}
+
+		/**
+		 * Fires after activation emails are resent.
+		 *
+		 * @since 2.0.0
+		 *
+		 * @param array $signup_ids Array of IDs to resend activation emails to.
+		 * @param array $result     Updated metadata related to activation emails.
+		 */
+		do_action( 'bp_core_signup_after_resend', $signup_ids, $result );
+
+		/**
+		 * Filters the result of the metadata for signup activation email resends.
+		 *
+		 * @since 2.0.0
+		 *
+		 * @param array $result Updated metadata related to activation emails.
+		 */
+		return apply_filters( 'bp_core_signup_resend', $result );
+	}
+
+	/**
+	 * Check if an activation email can be resent.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @param BP_Signup $signup The signup object.
+	 * @return bool
+	 */
+	public static function resend_activation( $signup ) {
```

Comment:
The function name makes this sound like an action and not a check. Maybe `allow_activation_resend()` would be clearer (and less close to `resend()`)?

Reply:
Agree! Will update.